inst/tests/tests.Rraw

```diff
@@ -21112,3 +21112,8 @@ test(2309.09, as.data.table(df, keep.rownames=TRUE), data.table(rn = c("a","b"),
 as.data.frame.no.reset = function(x) x
 DF = structure(list(a = 1:2), class = c("data.frame", "no.reset"), row.names = c(NA, -2L))
 test(2310.01, as.data.table(DF), data.table(a=1:2))
+
+# memrecycle() did not consider string encodings for factor levels #6886
+DT = data.table(factor(rep(enc2utf8("\uf8"), 3)))
+DT[1, V1 := iconv(levels(V1), from = "UTF-8", to = "latin1")]
```

Comment:
this test is _somewhat_ confusing, since the end result is basically "do nothing", maybe it'll help to be more practical?

e.g. `new_val = factor('a', levels = c('a', iconv(...)))`, then check that all we get in the update is an `a` level?

Re-reading the issue, I see your example is actually quite true to the original bug report. So maybe add this second example as another test instead?

Reply:
It's also a good idea to test both scenarios: assigning from a string and assigning from a factor.